internal/docstore/doc.go

```diff
@@ -13,38 +13,142 @@
+// for managing your initialization code.
+//
+// Alternatively, you can construct a *Collection via a URL and the OpenCollection
+// function in this package. See https://gocloud.dev/concepts/urls for more
+// information.
+//
+//
+// Documents
+//
+// Docstore allows you to represent documents as either map[string]interface{} or
+// struct pointers. Using structs is recommended, because it enforces some structure
+// on your data. Docstore mimics the encoding/json package in its treatment of
+// structs: by default, a struct's exported fields are the fields of the document.
+// You can alter this default mapping by using a struct tag beginning with
+// "docstore:". Docstore struct tags support renaming, omitting fields
+// unconditionally, or omitting them only when they are empty, exactly like
+// encoding/json.
+//
+//
+// Representing Data
+//
+// The value stored in a document field can be any of a wide range of types. All
+// primitive types except for complex numbers are supported, as well as slices and
+// maps (the map key type must be a string, an integer, or a type that implements
+// encoding.TextMarshaler). In addition, any type that implements
+// encoding.BinaryMarshaler or encoding.TextMarshaler is permitted. This set of types
+// closely matches the encoding/json package.
+//
+// Times deserve special mention. Docstore can store and retrieve values of type
+// time.Time, with two caveats. First, the timezone may not be preserved. Second,
+// Docstore guarantees only that time.Time values are represented to millisecond
+// precision. Many providers will do better, but if you need to be sure that times
+// are stored with nanosecond precision, convert the time.Time to another type before
+// storing, and re-create when you retrieve it. For instance, you could call the
+// time's UnixNano method to get an int64, and get the original time back (in the
+// local timezone) with the time.Unix function.
+//
+//
+// Keys
+//
+// The key of a docstore document is some function of its contents, usually a field.
+// Keys never appear alone in the docstore API, only as part of a document. For
+// instance, to retrieve a document by key, you pass the Get method a document whose
+// key field is populated, and docstore populates the rest of that argument with the
+// stored document.
+//
+// When you open a collection using one of the subpackages' OpenCollection methods or
+// a URL, you specify how to extract the key from a document.
+// Usually, you provide the name of the key field, as in the example above:
+//
+//   coll, err := memdocstore.OpenBucket("SSN", nil)
+//
+// Here, the "SSN" field of the document is used as the key. Some providers let you
+// supply a function to extract the key from the document, which can be useful if the
+// key is composed of more than field.
+//
+//
+// Actions
+//
+// Docstore supports six actions on documents:
+// - Get retrieves a document.
```

Comment:
I suggest you run `godoc` locally already and make sure the formatting of this looks OK. I think you might need to indent these by a space to make sure they are on separate lines...

Reply:
Fixed.